bytecomp/matching.ml

```diff
@@ -2225,163 +2568,190 @@ let mk_failaction_pos partial seen ctx defs  =
     None,fail,jmps
   end else begin (* Too many non-matched constructors -> reduced information *)
     if dbg then eprintf "POS->NEG!!!\n%!" ;
-    let fail,jumps =  mk_failaction_neg partial ctx defs in
+    let fail,jumps =  mk_failaction_neg loc partial ctx defs in
     if dbg then
       eprintf "FAIL: %s\n"
         (match fail with
         | None -> "<none>"
-        | Some lam -> string_of_lam lam) ;
+        | Some block -> string_of_block block) ;
     fail,[],jumps
   end
 
-let combine_constant loc arg cst partial ctx def
-    (const_lambda_list, total, _pats) =
-  let fail, local_jumps =
-    mk_failaction_neg partial ctx def in
+type combined = Lambda.block * jumps_union
+
+type constant_cases =
+  ((Location.t * Asttypes.constant) * Lambda.block) list
```

Comment:
This is `const_with_loc cases`. A generic `const located cases` could be nice -- they are other `Location.t * 'a` uses for the same reason below.

Reply:
I've added a new parameterised type.